pkg/cloud/aws/services/userdata/node.go

```diff
@@ -39,6 +39,30 @@ nodeRegistration:
 EOF
 
 kubeadm join --config /tmp/kubeadm-node.yaml
+`
+
+	nodeCloudInit = `{{.Header}}
+write_files:
+-   path: /tmp/kubeadm-node.yaml
+    owner: root:root
+    permissions: '0644'
+    content: |
+      ---
+      apiVersion: kubeadm.k8s.io/v1beta1
+      kind: JoinConfiguration
+      discovery:
+        bootstrapToken:
+          token: "{{.BootstrapToken}}"
+          apiServerEndpoint: "{{.ELBAddress}}:6443"
+          caCertHashes:
+            - "{{.CACertHash}}"
+      nodeRegistration:
+        name: {{ "{{ ds.meta_data.hostname }}" }}
+        criSocket: /var/run/containerd/containerd.sock
+        kubeletExtraArgs:
+          cloud-provider: aws
+runcmd:
+  - kubeadm join --config /tmp/kubeadm-node.yaml
```

Comment:
We deploy a version of cloud-init that contains a kubeadm module: https://github.com/detiber/cloud-init/blob/cluster-api/cloudinit/config/cc_kubeadm.py

Reply:
@detiber is the module deployed into the AMIs today?

Reply:
It is, it's added as part of the image baking process.

Reply:
Great, I like this solution much more than `runcmd`, will make a change. Thanks for the feedback.

Reply:
Looks like this `kubeadm` module is not baked into the AMIs:

```bash
root@ip-10-0-0-105:~# ls -la /usr/lib/python3/dist-packages/cloudinit/config/cc_kube*
ls: cannot access '/usr/lib/python3/dist-packages/cloudinit/config/cc_kube*': No such file or directory
root@ip-10-0-0-105:~# ls -la /usr/lib/python3/dist-packages/cloudinit/config/cc_puppet*
-rw-r--r-- 1 root root 7910 Oct  2 20:52 /usr/lib/python3/dist-packages/cloudinit/config/cc_puppet.py
root@ip-10-0-0-105:~#
```

Reply:
It looks like the `cloud-init` is not overwritten with the PPA package, hence the plugin `kubeadm` is not present. I will test CentOS and Amazon Linux tomorrow and create PR to fix this issue.